Flutter/pdf-viewer/form-filling.md

```diff
@@ -0,0 +1,193 @@
+
+* Text box. 
+* Checkbox. 
+* Radio button. 
+* Combo box. 
+* Signature. 
+
+## Save form data 
+
+You can save the modified form fields data by calling the `saveDocument` method. Refer to the following code example. 
+
+{% tabs %}
+{% highlight dart hl_lines="15 16" %}
+
+final PdfViewerController _pdfViewerController = PdfViewerController(); 
+ 
+@override 
+Widget build(BuildContext context) { 
+  return Scaffold( 
+    appBar: AppBar( 
+      title: const Text('Syncfusion Flutter PDF Viewer'), 
+      actions: <Widget>[ 
+        IconButton( 
+          icon: const Icon( 
+            Icons.save, 
+            color: Colors.white, 
+          ), 
+          onPressed: () async { 
+            final List<int> savedBytes = 
+                await _pdfViewerController.saveDocument(); 
+          }, 
+        ), 
+      ], 
+    ), 
+    body: SfPdfViewer.asset( 
+      'assets/form_document.pdf', 
+      controller: _pdfViewerController, 
+    ), 
+  ); 
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+N> When the `saveDocument` method is called, the document will be automatically reloaded after the save in the viewer. Also, the signature field will be flattened on save irrespective to the `PdfFlattenOption` provided.   
+
+### Flattening the form data on save 
+
+Flattening PDF form is a process of removing the form fields in the PDF document, thereby rendering the form fields appearance and its content in the page graphics. This will avoid the PDF form being edited in any device. Flutter PDF Viewer supports flattening the PDF form when saving. You can perform this action by setting the `PdfFlattenOption` to `PdfFlattenOption.formFields` in the `saveDocument` method.  
+
+By default, the `PdfFlattenOption` will be `PdfFlattenOption.none`, which means the form fields (except signature fields) can be edited after save.  
+
+Refer to the following code example. 
+
+{% tabs %}
+{% highlight dart hl_lines="15 16" %}
+
+final PdfViewerController _pdfViewerController = PdfViewerController(); 
+ 
+@override 
+Widget build(BuildContext context) { 
+  return Scaffold( 
+    appBar: AppBar( 
+      title: const Text('Syncfusion Flutter PDF Viewer'), 
+      actions: <Widget>[ 
+        IconButton( 
+          icon: const Icon( 
+            Icons.save, 
+            color: Colors.white, 
+          ), 
+          onPressed: () async { 
+            final List<int> savedBytes = await _pdfViewerController 
+                .saveDocument(flattenOption: PdfFlattenOption.formFields); 
+          }, 
+        ), 
+      ], 
+    ), 
+    body: SfPdfViewer.asset( 
+      'assets/form_document.pdf', 
+      controller: _pdfViewerController, 
+    ), 
+  ); 
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Importing and Exporting form data 
+
+The Flutter PDF viewer allows the users to import and export form data to and from the PDF documents. The import and export of form data supports the following extensions. 
```

Comment:
The Flutter PDF viewer allows users to import and export form data to and from PDF documents. The import and export of form data support the following extensions.